src/main/java/org/assertj/core/api/AbstractFileAssert.java

```diff
@@ -939,6 +939,53 @@ public SELF isDirectoryContaining(String syntaxAndPattern) {
     return myself;
   }
 
+  /**
+   * Verify that the actual {@code File} is a directory containing recursively at least one file matching the given {@code String}
+   * interpreted as a path matcher (as per {@link FileSystem#getPathMatcher(String)}).
+   * <p>
+   * Note that the actual {@link File} must exist and be a directory.
+   * <p>
+   * Given the following directory structure:
+   * <pre><code class="text"> /root/
+   * /root/foo-dir/
+   * /root/foo-dir/foobar-dir
+   * /root/foo-dir/foobar-dir/foo-file-1.ext
+   * /root/foo-file-2.ext</code></pre>
```

Comment:
I know the other examples are like this one but I was thinking using a unix tree like syntax might be nice, ex:
```
root        
|-- foo-dir     
|    |-- foobar-dir  
|    |-- foobar-dir  
|         |-- foo-file-1.ext
|-- foo-file-2.ext
```            
WDYT?

Reply:
Excellent idea, done :)

Reply:
Not done for the unit test class while that would be not bad either.